### PR TITLE
Management of non Docker or S2I build types

### PR DESCRIPTION
**Pull request by sabre1041**
Addresses how to handle other build types that are not compatible with producing output images

### CODE REVIEW

File: pkg/webhook/webhook.go [hunk 1]
```diff
@@ -124,7 +124,7 @@ func getAdmissionResponseForBuild(ar *v1beta1.AdmissionReview, quayIntegration *
 
 	quayRegistryHostname, err := quayIntegration.GetRegistryHostname()
 
-	if build.Spec.CommonSpec.Output.To.Kind != "ImageStreamTag" {
+	if buildv1.JenkinsPipelineBuildStrategyType == build.Spec.Strategy.Type || build.Spec.CommonSpec.Output.To == nil || build.Spec.CommonSpec.Output.To.Kind != "ImageStreamTag" {
```
Comment (dmesser) [inline pkg/webhook/webhook.go]: If I read the [docs](https://docs.openshift.com/container-platform/4.6/builds/managing-build-output.html) correct it seems like only the strategies (`build.spec.strategy.type`) of type `sourceStrategy` or `dockerStrategy` or `customStrategy` qualify for build outputs to be images. So maybe a reverse check on them is more future-proof than intercepting Jenkins builds. To be safe I'd also check if the `spec.output` element exists.

**Review by dmesser — CHANGES_REQUESTED** [review]

File: pkg/webhook/webhook.go (hunk 1 above)
Comment (sabre1041) [reply pkg/webhook/webhook.go]: Updated the PR

File: pkg/webhook/webhook.go [hunk 2]
```diff
@@ -124,7 +124,7 @@ func getAdmissionResponseForBuild(ar *v1beta1.AdmissionReview, quayIntegration *
 
 	quayRegistryHostname, err := quayIntegration.GetRegistryHostname()
 
-	if buildv1.JenkinsPipelineBuildStrategyType == build.Spec.Strategy.Type || build.Spec.CommonSpec.Output.To == nil || build.Spec.CommonSpec.Output.To.Kind != "ImageStreamTag" {
+	if (build.Spec.Strategy.DockerStrategy == nil && build.Spec.Strategy.SourceStrategy == nil) || build.Spec.CommonSpec.Output.To.Kind != "ImageStreamTag" {
```
Comment (dmesser) [inline pkg/webhook/webhook.go:127]: What about custom build strategies? Other than that, assuming that Docker and S2I strategies mandate that `spec.ouput.to` is not NIL I think this lgtm

Comment (sabre1041) [reply pkg/webhook/webhook.go:127]: Custom Build strategies is a big question on whether it should be supported which is why I went down my original approach to really only exclude `JenkinsPipeline` and then do checks on whether certain fields are present.

`Output` is not a pointer so there will not be an issue of hitting a NPE

Comment (dmesser) [reply pkg/webhook/webhook.go:127]: Ack on custom builds not being supported. I think we should not break the system when we encounter such a build. People could be mixing several types of build configs in a namespace. Not supporting it by gracefully ignoring those seems the best way to me.

Comment (sabre1041) [reply pkg/webhook/webhook.go:127]: As the logic stands now, custom strategies would not have their output image destinations mutated and would be ignored. Is that the expectation that we would want to deliver?

Comment (dmesser) [reply pkg/webhook/webhook.go:127]: Sounds good to me for now.

**Pull request merged**